restli-client/src/main/java/com/linkedin/restli/client/RestClient.java

```diff
@@ -1108,6 +1111,9 @@ public void onSuccess(ProtocolVersion protocolVersion)
         if (mappingResults == null || mappingResults.getMappedKeys().isEmpty())
         {
           // Strategy returns null URIMappingResult or empty mapped hosts, assuming no scatter is needed
+          LOG.error("ScatterGatherStrategy cannot map URIs for request: " + request
+              + ", requestContext: " + requestContext
+              + ", ScatterGatherStrategy needScatterGatherStrategy value: " +  strategy.needScatterGather(request));
```

Comment:
Is it possible for `strategy` to be null?

Reply:
I don't think strategy can be null. Look at line 1100, if strategy is null, it would not reach here.